python/services/evaluationservice/dmod/evaluationservice/worker.py

```diff
@@ -0,0 +1,242 @@
+        self.__evaluation_name = parameters.name
+        self.__verbosity = Verbosity[verbosity]
+        self.__start_delay = int(parameters.delay) if parameters.delay else 0
+        self.__format = parameters.format
+
+
+def evaluate(evaluation_id: str, definition_json: str, arguments: Arguments = None) -> dict:
+    evaluation_id = evaluation_id.replace(" ", "_")
+
+    redis_host = arguments.redis_host if arguments else None
+    redis_port = arguments.redis_port if arguments else None
+    redis_password = arguments.redis_password if arguments else None
+    delay_seconds = int(arguments.start_delay) if arguments else 0
+
+    if arguments:
+        verbosity = arguments.verbosity
+    else:
+        verbosity = Verbosity[os.environ.get("EVALUATION_VERBOSITY", "NORMAL").upper()]
+
+    should_publish = verbosity >= Verbosity.NORMAL
+
+    sleep(delay_seconds)
+
+    communicators = utilities.get_communicators(
+        communicator_id=evaluation_id,
+        verbosity=verbosity,
+        host=redis_host,
+        port=redis_port,
+        password=redis_password
+    )
+
+    error_key = utilities.key_separator().join([utilities.redis_prefix(), evaluation_id, "ERRORS"])
+    message_key = utilities.key_separator().join([utilities.redis_prefix(), evaluation_id, "MESSAGES"])
+
+    communicators.update(
+        created_at=utilities.now().strftime(COMMON_DATETIME_FORMAT),
+        failed=False,
+        complete=False,
+        error_key=error_key,
+        message_key=message_key
+    )
+
+    try:
+        definition = json.loads(definition_json)
+    except Exception as exception:
+        message = "The evaluation instructions could not be loaded"
+        communicators.error(message, exception, publish=should_publish)
+        communicators.error(definition_json, None, publish=should_publish)
+        communicators.update(failed=True)
+        communicators.sunset(60*3)
+        raise exception
+
+    write_results = {
+        "success": False
+    }
+
+    try:
+        evaluator = Evaluator(definition, communicators=communicators, verbosity=verbosity)
+        communicators.info(f"starting {evaluation_id}", publish=should_publish)
+        results = evaluator.evaluate()
+        communicators.info("Result: {:.2f}%".format(results.grade), publish=should_publish)
+        communicators.info(f"{evaluation_id} complete; now writing results")
+        write_results = writing.write(evaluation_id=evaluation_id, results=results, output_format=arguments.format)
+        communicators.info(f"Data from {evaluation_id} was written.")
+    except Exception as e:
+        communicators.error(f"{e.__class__.__name__}: {e}", e, publish=should_publish)
+        communicators.update(failed=True)
+        communicators.sunset(60*3)
+    finally:
+        communicators.update(complete=True)
+        communicators.info(f"{evaluation_id} is complete", publish=should_publish)
+
+    return write_results
+
+
+def main(arguments: Arguments = None):
+    """
```

Comment:
Related to the comments above about the general design of DMOD services and possibly needing to move out the GUI stuff to a separate place, I think we'd want to alter this as well.  I'd suggest having everything start in the package _\_\_main\_\_.py_, though that could have more args that direct execution to this module.

Reply:
`worker.py` needs to stay away from the core process. Maybe there's more flexibility now that RQ isn't being used, but it needs to be run out of process with access to _some_ common values (like logging configuration and application values), but no connection to the service at large.